lib/skate_web/controllers/redirect_controller.ex

```diff
@@ -0,0 +1,19 @@
+defmodule SkateWeb.Redirect do
+  @moduledoc """
+  Controller that redirects to the `:external` `opts` when `call/2`-ed.
+
+  ---
+
+  Reference: https://www.viget.com/articles/how-to-redirect-from-the-phoenix-router/
+  """
+  import Phoenix.Controller, only: [redirect: 2]
+
+  @spec init(Keyword.t()) :: Keyword.t()
+  def init([external: _] = opts), do: opts
+  def init(_default), do: raise("Missing required external: option in redirect")
+
+  @spec call(Plug.Conn.t(), Keyword.t()) :: Plug.Conn.t()
+  def call(conn, external: url) do
```

Comment:
**suggestion:** related to your other comment about avoiding open redirects, I have a thought on another approach that might be slightly more watertight. Instead of passing a full external URL in, the `Redirect` controller could instead take a single atom argument (say `:aup` in this case). Then, this module could have a map of atoms to URLs for each of the static files that we want to be able to redirect our users to. The controller could then look up the URL from this map and throw an error if the corresponding value is null, which wouldn't allow us to redirect to any URLs not explicitly listed. Thoughts?